### ISSUE TITLE
Scratch buffer contents is lost by accident

### ISSUE THREAD

**Issue by lrworth** [mawww/kakoune#2759]
This put me off Kakoune for a long time:

1. `kak`
2. Enter lots of text, intended to be the contents of a new file.
3. Accidentally `:q`

And you lose the whole lot! Now I understand that \*scratch* is designed to be lost, but this will definitely burn people, as most other editors don't delete data this readily. Emacs solves this by adding text at the beginning of \*scratch*:
```
;; This buffer is for notes you don't want to save, and for Lisp evaluation.                                                                                                                          
;; If you want to create a file, visit that file with C-x C-f,                                                                                                                                        
;; then enter the text in that file's own buffer.                                                                                                                                                     
```

But I'd suggest copying vim, and opening a non-scratch unsaved buffer by default.

**Comment by ekie** [comment]
I like the current design and get annoyed when using vi(m) and cannot just `:q` in some usecases. however you can set an alias in your shell: `alias k='kak newfile'`

Maybe going the emacs route is a viable option to improve the user experience.

related #2360 & #1496

**Comment by andreyorst** [comment]
I'd also wanted `*scratch*` to be something like a buffer for `kakscript` evaluation, similar to Emacs' `*scratch*`. I really love that in Emacs you can fire up a scratch buffer, write some Emacs Lisp in it, and evaluate it. This is certainly doable with Kakoune's `source %val{buffile}` but it requires a file to do evaluation.

**Comment by alexherbo2** [comment]
``` kak
map global user s -docstring 'Evaluate the main selection' ':<c-r>.<ret>'
map global user S -docstring 'Evaluate the current buffer' '%:<c-r>.<ret>'
```

**Comment by andreyorst** [comment]
Thanks, @alexherbo2. That's a nice addition to my configuration

**Comment by alexherbo2** [comment]
@mawww I preferred the scratch buffer being empty.

**Comment by alexherbo2** [comment]
``` kak
hook -always -once global WinCreate \Q*scratch* %{
  hook -always -once window InsertEnd '' %{
    info -title 'Scratch' 'This is a scratch buffer which won’t be automatically saved.  Use it for notes or open a file buffer with the :edit command'
  }
}
```

**Comment by andreyorst** [comment]
can I delete `*scratch*` automatically if no changes were made to it and another file was opened?

1. Start kakoune with no arguments
    - buffers: `*scratch*`, `*debug*`
2. Open file
    - buffers: `file`, `*debug*`

Or:
1. Start kakoune with no arguments
    - buffers: `*scratch*`, `*debug*`
2. Write something to scratch and open file.
    - buffers: `file`, `*scratch*`, `*debug*`

**Comment by occivink** [comment]
@andreyorst I've got this in my kakrc that pretty much does it
```
# delete the *scratch* buffer as soon as another is created, but only if it's empty
hook global BufCreate '^\*scratch\*$' %{
    hook -once -always global BufCreate '^(?!\*scratch\*).*$' %{
        try %{
            # throw if the buffer has more than one character
            exec -buffer *scratch* 'L<a-K>..<ret>'
            db *scratch*
        }
    }
}
```

Likely needs to be updated now that `*scratch*` contains default text

**Comment by alexherbo2** [comment]
The scratch buffer being populated with text feels like dirty.

**Comment by andreyorst** [comment]
> The scratch buffer being populated with text feels like dirty.

I think that the default text should be configured via option, much like in Emacs. I don't use default text there, but some people may do.

**Comment by alexherbo2** [comment]
I don’t agree on the direction taken to resolve this issue.

**Comment by mawww** [comment]
@alexherbo2 I agree it is not very satisfying, there is a already a `[scratch]` marker in the mode line, but this is an issue that comes back regularly and I figured that would make new comers more aware of the semantics of scratch buffers in Kakoune.

Happy to revisit if we find a better way to make scratch buffer behaviour more obvious (I think the scratch buffer behaviour is sane, but newcomers do not know what to expect from them and get confused that what they have typed in it gets lost with no warning).

**Comment by alexherbo2** [comment]
@mawww How about https://github.com/mawww/kakoune/issues/1496#issuecomment-473907027

**Comment by JJK96** [comment]
> I think that the default text should be configured via option, much like in Emacs. I don't use default text there, but some people may do.

I agree, this way it is on by default, but easy to turn off, just like a lot of other settings (clippy comes to mind)

**Comment by GavinSmith0123** [comment]
I've just been hit by this issue. Opened up a new kakoune instance with the aim of writing a short note that I was going to copy into a web browser, then my short note ended up taking longer to write than I expected. Without thinking I typed ":w\<ret>:q\<ret>" as this was muscle memory for me to save and exit kakoune. (I have mapped my Caps Lock key to `:` so this is quick for me to type.) I would have expected to have needed to type ":q!" instead to exit kakoune with an unsaved scratch buffer.

The lesson I take from this is never to use the scratch buffer for any purpose.
